### ISSUE TITLE
Wrong resolution upon 2nd pipenv install

### ISSUE THREAD

**Issue by zsimic** [pypa/pipenv#1189]
##### Describe your environment

1. macOS 10.12.6
1. Python version: any python version, bug seen with 2.7 and 3.6
1. Pipenv version: 9.0.0

##### Expected result

Expecting `pipenv install` to yield same result consistently, in this particular case `setuptools 33.1.1` should be resolved (due to pex constraint)

##### Actual result

pipenv resolves `setuptools 33.1.1` on 1st installation, but not on 2nd

##### Steps to replicate

You can reproduce like so (in any folder, using a scratch one here).

First time around, everything looks good (pex wants `setuptools<34`), note the minor annoying question **"Did you mean `pexpect`?"**:

```
pipenv --rm ; rm -f Pipfile* ; pipenv install pex ; pipenv install pytest --dev ; pipenv graph
...
Did you mean pexpect? [Y/n]: n
Installing pex…
...
pex==1.2.15
  - setuptools [required: >=20.3,<34.0, installed: 33.1.1]
...
pytest==3.3.1
  - setuptools [required: Any, installed: 33.1.1]
...
```

Now delete venv and install again -> wrong setuptools is resolved:

```
pipenv --rm ; pipenv install ; pipenv graph
...
pex==1.2.15
  - setuptools [required: >=20.3,<34.0, installed: 38.2.4]
...
pytest==3.3.1
  - setuptools [required: Any, installed: 38.2.4]
...
```

Also note that `pipenv install` acted here as-if I state `--dev`... wasn't expecting `pytest` to be installed here (was expecting it to be installed if I stated `--dev`).

**Comment by zsimic** [comment]
Found a workaround for this, by running:
```
pipenv --rm ; pipenv install
pipenv run pip install -e .
```

`pip install` notices the disparity and fixes it.

Can't do `pipenv install '-e .'`: that doesn't help (setuptools version is still broken).

**Comment by techalchemy** [comment]
@zsimic thanks for the report. Just want you to know you are not crazy, this is a real issue introduced in 9.0 which always installs dev packages by default. It is fixed in #1173 and will be released shortly. I’m going to close this for now, but sorry for the inconvenience!
